src/pip/_internal/index/collector.py

```diff
@@ -518,16 +518,19 @@ def __init__(
         self,
         session: PipSession,
         search_scope: SearchScope,
+        index_lookup: dict = None,
```

Comment:
```suggestion
        index_lookup: Optional[Dict[str, List[str]] = None,
```

Reply:
I think the way I am using it right now its really a:
` Optional[Dict[str, str]]` but I could see value maybe in `Optional[Dict[str, List[str]]]` if supporting the ability to restrict a package to a list of indexes, which I think might make sense for someone though on average the requirement is to pin a package to a specific index for reasons such as:

- The package we want exists in a specific index but confusion packages exist with the same name in other pypis and so it resolves to the wrong package when not strictly version pinned (or worse, the same version numbers exist in both places but are two different packages).
- The package is very large and we want to be as specific as possible about the index so that we don't download many extra GB of files from say alternate pytorch indexes trying to resolve and install the package.